python/GafferSceneTest/IECoreScenePreviewTest/CapturingRendererTest.py

```diff
@@ -165,5 +168,390 @@ def testDeformingObject( self ) :
 		self.assertEqual( c.capturedSamples(), [ sphere1, sphere2 ] )
 		self.assertEqual( c.capturedSampleTimes(), [ 1, 2 ] )
 
+	class TestProcedural( GafferScene.Private.IECoreScenePreview.Procedural ) :
+
+		def __init__( self, n ) :
+
+			GafferScene.Private.IECoreScenePreview.Procedural.__init__( self )
+			self.n = n
+
+		def render( self, renderer ) :
+
+			attributes = renderer.attributes( IECore.CompoundObject( { "a" : IECore.FloatData( 12 ) } ) )
+			self.o = renderer.object( "a", IECoreScene.SpherePrimitive( 42 ), attributes )
+			self.o.transform( imath.M44f().translate( imath.V3f( 3, 0, 0 ) ) )
+
+	class TestRecursiveProcedural( GafferScene.Private.IECoreScenePreview.Procedural ) :
+
+		def __init__( self ) :
+
+			GafferScene.Private.IECoreScenePreview.Procedural.__init__( self )
+
+		def render( self, renderer ) :
+
+			attributes = renderer.attributes( IECore.CompoundObject( { "b" : IECore.FloatData( 7 ) } ) )
+			self.p = renderer.object( "b", CapturingRendererTest.TestProcedural( 1 ), attributes )
+			self.p.transform( [
+					imath.M44f( 0, 1, 0, 0,  -1, 0, 0, 0,  0, 0, 1, 0,  0, 0, 0, 1 ),
+					imath.M44f( 0, -1, 0, 0,  1, 0, 0, 0,  0, 0, 1, 0,  0, 0, 0, 1 )
+				], [ 0, 1 ]
+			)
+
+			attributes = renderer.attributes( IECore.CompoundObject() )
+			self.o = renderer.object( "x", IECoreScene.SpherePrimitive( 37 ), attributes )
+
+	ExpandedCapture = collections.namedtuple( 'ExpandedCapture',
+		[ 'capturedSamples', 'capturedSampleTimes', 'capturedTransforms', 'capturedTransformSampleTimes',
+		'capturedAttributes', 'capturedLinks' ]
+	)
+
+	@staticmethod
+	def expandCapturingRenderer( capturingRenderer, expandProcedurals = False ):
```

Comment:
Can we make `ExpandedCapture` and `expandCapturingRenderer` private? They only seem to be used in here, and we can easily make them public again if we have any need.